test-data/unit/check-unreachable-code.test

```diff
@@ -1446,3 +1446,11 @@ def f() -> None:
     Foo()['a'] = 'a'
     x = 0 # This should not be reported as unreachable
 [builtins fixtures/exception.pyi]
+
+[case testIntentionallyEmptyGenerator]
+# flags: --warn-unreachable
+from typing import Generator
+
+def f() -> Generator[None, None, None]:
+    return
```

Comment:
`return None` has the same meaning and is just stylistically different. Let's test it as well. 

Other than that - LGTM!

Reply:
Ditto for `yield None` - do we want to support all of the forms?

Reply:
It won't hurt, I guess :)

Reply:
7871b8d549793c626e23699b182f9cdec8570f23